mongo/filterstore.go

```diff
@@ -185,10 +185,32 @@ func (s *FilterStore) AddFilterDimension(dimension *models.AddDimension) error {
 	session := s.Session.Copy()
 	defer session.Close()
 
-	queryFilter := bson.M{"filter_job_id": dimension.FilterID}
+	list, err := s.GetFilterDimensions(dimension.FilterID)
+	if err != nil && err != errNotFound {
+		return errBadRequest
+	}
+
 	url := fmt.Sprintf("%s/filter/%s/dimensions/%s", s.host, dimension.FilterID, dimension.Name)
-	d := models.Dimension{Name: dimension.Name, Options: dimension.Options, DimensionURL: url}
-	update := bson.M{"$addToSet": bson.M{"dimensions": d}}
+	d := models.Dimension{Name: dimension.Name, Options: dimension.Options, URL: url}
+
+	var found bool
+	var index int
+	for i, item := range list {
+		if item.Name == d.Name {
+			found = true
+			index = i
```

Comment:
You can replace dimension here and then there will be no need for the variable index.

Something like this:
```
for i, item := range list {
  if item.Name == d.Name {
    list[i] = d
    found = true
    break
  }
}

if !found {
  list = append(list, d)
}
```